amp/amp-service-worker-offline-commenting.js

```diff
@@ -0,0 +1,44 @@
+/* global ERROR_MESSAGES */
+{
+	const queue = new wp.serviceWorker.backgroundSync.Queue( 'amp-wpPendingComments' );
+	const errorMessages = ERROR_MESSAGES;
+
+	const commentHandler = ( { event } ) => {
+		const clone = event.request.clone();
+		return fetch( event.request )
+			.then( ( response ) => {
+				// @todo Make sure that 409 etc. error still work as expected.
+				return response;
+			} )
+			.catch( () => {
+				const bodyPromise = clone.blob();
+				bodyPromise.then(
+					function( body ) {
+						const request = event.request;
+						const req = new Request( request.url, {
+							method: request.method,
+							headers: request.headers,
+							mode: 'same-origin',
+							credentials: request.credentials,
+							referrer: request.referrer,
+							redirect: 'manual',
+							body: body
+						} );
+
+						// Add request to queue. @todo Replace when upgrading to Workbox v4!
+						queue.addRequest( req );
+					}
+				);
+
+				// @todo That's not actually working yet.
```

Comment:
I think you have to move this into the `catch` function, like so:

```js
const bodyPromise = clone.blob();
bodyPromise.then(
	function( body ) {
		const request = event.request;
		const req = new Request( request.url, {
			method: request.method,
			headers: request.headers,
			mode: 'same-origin',
			credentials: request.credentials,
			referrer: request.referrer,
			redirect: 'manual',
			body: body
		} );

		// Add request to queue. @todo Replace when upgrading to Workbox v4!
		queue.addRequest( req );

		const body = JSON.stringify( { 'error': errorMessages.comment } );
		return new Response( body, {} );
	}
);
```

You only want it to be sent back in the case of an error (`catch`) scenario. Also, you should wait to send it until the `bodyPromise` resolves, as otherwise I believe it the callback can be killed.